internal/binutils/binutils.go

```diff
@@ -140,7 +144,72 @@ func initTools(b *binrep, config string) {
 		b.addr2line, b.addr2lineFound = findExe("gaddr2line", append(paths["addr2line"], defaultPath...))
 	}
 	b.nm, b.nmFound = findExe("nm", append(paths["nm"], defaultPath...))
-	b.objdump, b.objdumpFound = findExe("objdump", append(paths["objdump"], defaultPath...))
+	// Look for llvm-objdump and objdump
+	b.objdump, b.llvmObjdumpFound, b.objdumpFound = FindObjdump(append(paths["llvm-objdump"], append(paths["objdump"], append(paths["gobjdump"], defaultPath...)...)...))
+}
+
+// FindObjdump finds and returns preferred objdump binary. Order of preference is: llvm-objdump, objdump.
+// On MacOS only, also looks for gobjdump with least preference.
+func FindObjdump(paths []string) (string, bool, bool) {
+	objdumpNames := []string{"llvm-objdump", "objdump"}
+	if runtime.GOOS == "darwin" {
+		objdumpNames = append(objdumpNames, "gobjdump")
+	}
+
+	for _, objdumpName := range objdumpNames {
+		if objdump, objdumpFound := findExe(objdumpName, paths); objdumpFound {
+			cmdOut, err := exec.Command(objdump, "--version").Output()
+			if err != nil {
+				continue
+			}
+			if llvmObjdumpFound := isLlvmObjdump(string(cmdOut)); llvmObjdumpFound {
+				return objdump, llvmObjdumpFound, false
+			}
+			if buObjdumpFound := isBuObjdump(string(cmdOut)); buObjdumpFound {
+				return objdump, false, buObjdumpFound
+			}
+		}
+	}
+	return "", false, false
+}
+
+func isLlvmObjdump(output string) bool {
+	fields := objdumpLLVMVerRE.FindStringSubmatch(output)
+	if len(fields) == 6 {
```

Comment:
```
if len(fields) != 6 {
  return false
}
```

and unindent.

https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow